src/zenml/integrations/mlflow/steps/mlflow_deployer.py

```diff
@@ -134,25 +132,48 @@ def mlflow_model_deployer_step(
         service = cast(MLFlowDeploymentService, existing_services[0])
 
     # check for conditions to deploy the model
-    if not config.model_uri:
-        # an MLflow model was not found in the current run, so we simply reuse
-        # the service created during the previous step run
+    if not model_uri:
+        # an MLflow model was not trained in the current run, so we simply reuse
+        # the currently running service created for the same model, if any
         if not existing_services:
-            raise RuntimeError(
+            logger.warning(
                 f"An MLflow model with name `{config.model_name}` was not "
-                f"trained in the current pipeline run and no previous "
-                f"service was found."
+                f"logged in the current pipeline run and no running MLflow "
+                f"model server was found. Please ensure that your pipeline "
+                f"includes an `mlflow_enable` decorated step that trains a "
+                f"model and logs it to MLflow. This could also happen if "
+                f"re-training a model in the current pipeline run was skipped "
```

Comment:
Is there something missing between 'if' and 're-training'? Maybe it's missing a 'while'?

Reply:
you're right, it reads awkwardly, I'll rephrase it to be easier to read